src/main/java/edu/cornell/kfs/module/ar/service/impl/CuContractsGrantsInvoiceCreateDocumentServiceImpl.java

```diff
@@ -111,5 +118,52 @@ protected String findContractControlAccountNumber(List<InvoiceAccountDetail> det
     protected String findTitleFormatString() {
         return getConfigurationService().getPropertyValueAsString(CuArParameterConstants.CONTRACTS_GRANTS_INVOICE_DOCUMENT_TITLE_FORMAT);
     }
+    
+    /**
+     * Added evaluation of account sub-fund not being an expenditure to existing base code evaluation
+     * of billing frequency and invoice document status when determining valid award accounts.
+     * Expenditure sub-funds associated to accounts to exclude are defined in CG system parameter
+     * CG_INVOICING_EXCLUDE_EXPENSES_SUB_FUNDS.
+     */
+    @Override
+    protected List<ContractsAndGrantsBillingAwardAccount> getValidAwardAccounts(
+            List<ContractsAndGrantsBillingAwardAccount> awardAccounts, ContractsAndGrantsBillingAward award) {
+        if (!ArConstants.BillingFrequencyValues.isMilestone(award) && !ArConstants.BillingFrequencyValues.isPredeterminedBilling(award)) {
+            List<ContractsAndGrantsBillingAwardAccount> validAwardAccounts = new ArrayList<>();
+            Set<Account> invalidAccounts = harvestAccountsFromContractsGrantsInvoices(getInProgressInvoicesForAward(award));
+
+            for (ContractsAndGrantsBillingAwardAccount awardAccount : awardAccounts) {
+                if (!invalidAccounts.contains(awardAccount.getAccount())) {
+                    if (verifyBillingFrequencyService.validateBillingFrequency(award, awardAccount) &&
+                        isNotExpenditureAccount(awardAccount)) {
```

Comment:
This line may need to be indented further, to avoid being in the same column as the code line below it.